rfcs/2022-02-11-responsive-and-adaptive-layout-strategy.md

```diff
@@ -0,0 +1,391 @@
+  - [How will we actually implement the approaches discussed above?](#how-will-we-actually-implement-the-approaches-discussed-above)
+    - [On Blade Code Side](#on-blade-code-side)
+    - [On Consumer Code Side](#on-consumer-code-side)
+    - [On Blade Design Side](#on-blade-design-side)
+    - [On Consumer Design Side](#on-consumer-design-side)
+- [Adoption strategy](#adoption-strategy)
+- [How do we educate people?](#how-do-we-educate-people)
+- [Open Questions](#open-questions)
+- [References](#references)
+
+# Summary
+
+To render content on different device screen resolutions there are different strategies. Modern web design gives us three options to use: Responsive, Adaptive, and Standalone Design, although standalone has fallen into disuse because of the overhead it comes with to maintain multiple codebase and their development lifecycle hence, we would just be focussing on Adaptive and Responsive.
+
+1. Responsive - Make the content fluid so it'll render based on the available space
+2. Adaptive - Make content targeted for specific screen sizes so it'll render exactly like that, regardless of the available space.
+
+This RFC will focus on what approach and strategies we shall take in our Design System.
+
+# Basic Example
+
+### Responsive Modal
+
+👇🏻 An example of a responsive design where we have modal which is responsive to the screen size and has a `maxWidth` defined for it to render on larger screen sizes, where on smaller screen sizes(mobile) it'll take `100%` of the width of the screen.
+
+<img alt="Responsive Modal" src="./images/responsive-adaptive-layout/responsive-modal.png">
+
+<br/>
+
+### Adaptive Search Field
+
+👇🏻 An example of a adaptive design where we have two implementation of a search field.
+
+1. Larger Screen Size - Search Input with Autosuggestion
+2. Smaller Screen Size - Search Field when clicked opens up a full screen modal which basically gives an optimised search experience on smaller screen size.
+
+<img alt="Responsive Modal" src="./images/responsive-adaptive-layout/adaptive-search-field.png">
+
+# Motivation
+
+### Why are we doing this?
+
+As of today, we don't have any strategy or approach in place for designing layouts for different screen sizes. Ensuring that elements fit within a page is not enough. There are different experiences that works for different screen sizes.
+
+Just going Responsive(Fluid) Design blindly is not correct. Imagine for example, you have different experiences(styles, javascript) for mobile vs desktop but if you just go blindly with responsive we might unnecessary download assets for all the screen sizes our application would run on. For example it doesn't makes sense to implement [this Search Field](#adaptive-search-field) as a responsive layout since it'll result in unnecessary download of the assets(modal for mobile) and hence hampering the experience and performance for the consumers.
```

Comment:
Improper capitalisation. Instead, it should be `Just going Responsive(fluid) design`